kvserver: TestSideloadingSideloadedStorage fails with RocksDB

Randomly when I was testracing the whole package:
```
--- FAIL: TestSideloadingSideloadedStorage (0.07s)
    --- FAIL: TestSideloadingSideloadedStorage/Disk (0.05s)
        replica_sideload_test.go:295: error truncating sideloaded storage: while purging "/tmp/TestSideloadingSideloadedStorage_Disk896201407/sideloading/r0XXXX/r1": IO error: file rmdir: /tmp/TestSideloadingSideloadedStorage_Disk896201407/sideloading/r0XXXX/r1: Directory not empty
            (1) attached stack trace
              | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*diskSideloadStorage).TruncateTo
              | 	/home/andrei/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_sideload_disk.go:241
              | github.com/cockroachdb/cockroach/pkg/kv/kvserver.testSideloadingSideloadedStorage.func7
              | 	/home/andrei/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_sideload_test.go:289
              | github.com/cockroachdb/cockroach/pkg/kv/kvserver.testSideloadingSideloadedStorage
              | 	/home/andrei/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_sideload_test.go:341
              | github.com/cockroachdb/cockroach/pkg/kv/kvserver.TestSideloadingSideloadedStorage.func2
              | 	/home/andrei/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_sideload_test.go:96
              | testing.tRunner
              | 	/home/andrei/go1.13/go/src/testing/testing.go:909
              | runtime.goexit
              | 	/home/andrei/go1.13/go/src/runtime/asm_amd64.s:1357
            Wraps: (2) 2 safe details enclosed
            Wraps: (3) while purging "/tmp/TestSideloadingSideloadedStorage_Disk896201407/sideloading/r0XXXX/r1"
            Wraps: (4) IO error: file rmdir: /tmp/TestSideloadingSideloadedStorage_Disk896201407/sideloading/r0XXXX/r1: Directory not empty
            Error types: (1) *withstack.withStack (2) *safedetails.withSafeDetails (3) *errutil.withMessage (4) *storage.Error
```

--- a/pkg/kv/kvserver/replica_sideload_test.go
+++ b/pkg/kv/kvserver/replica_sideload_test.go
@@ -290,8 +290,7 @@ func testSideloadingSideloadedStorage(
 		if err == nil {
 			t.Fatalf("sideloaded directory should not have been removable due to extra file %s", nonRemovableFile)
 		}
-		expectedTruncateError := "while purging %q: remove %s: directory not empty"
-		if err.Error() != fmt.Sprintf(expectedTruncateError, ss.(*diskSideloadStorage).dir, ss.(*diskSideloadStorage).dir) {
+		if !strings.HasSuffix(strings.ToLower(err.Error()), "directory not empty") {
 			t.Fatalf("error truncating sideloaded storage: %+v", err)
 		}
 		// Now remove extra file and let truncation proceed to remove directory.

